src/porepy/numerics/ad/operators.py

```diff
@@ -488,24 +487,25 @@ def _parse_readable(self, op: "Operator") -> str:
             operator_str = "*"
         elif tree.op == Operation.div:
             operator_str = "/"
-        elif tree.op in [Operation.evaluate, Operation.apply]:
-            # TODO: This has not really been tested.
+        # function evaluations have their own readable representation
+        elif tree.op == Operation.evaluate:
             is_func = True
-        else:
-            # TODO: This corresponds to unknown (to EK) cases.
-            print("Have not implemented string parsing of this operator")
+        # for unknown operations, 'operator_str' remains None
 
+        # error message for function evaluations
         if is_func:
-            # TODO: Not sure what to write here
-            msg = f"{child_str[0]} evaluated on ("
-            for child in range(1, len(child_str)):
-                msg += f"{child_str[child]}, "
-
+            msg = f"{child_str[0]}("
+            msg += ", ".join([f"{child}" for child in child_str[1:]])
             msg += ")"
             return msg
+        # if operation is unknown, a new error will be raised to raise awareness
+        elif operator_str is None:
+            msg = "UNKNOWN parsing of operation on: "
+            msg += ", ".join([f"{child}" for child in child_str])
+            raise NotImplementedError(msg)
```

Comment:
Hi @keileg ,
I have supplemented the docs now and introduced this NotImplementedError here for unknown operations.
This will give error message of the following form:
'During handling of the above exception, another exception occurred:' (raising errors in an except block)
So the actual error with the operation should not be covered up.